### PR TITLE
ddl: support create index with type and comment

### PR DESCRIPTION
**Pull request by jackysp**
1. Support create index with type and comment. TiDB only support alter table add index with type and comment. #3363
2. Fix missing information (type, comment) in show index from table. #3803 
PTAL @shenli @tiancaiamao @zimulala @XuHuaiyu

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@zimulala PTAL

File: ddl/index.go [hunk 1]
```diff
@@ -209,6 +210,13 @@ func (d *ddl) onCreateIndex(t *meta.Meta, job *model.Job) (ver int64, err error)
 			job.State = model.JobCancelled
 			return ver, errors.Trace(err)
 		}
+		if indexOption != nil {
+			indexInfo.Comment = indexOption.Comment
+			indexInfo.Tp = indexOption.Tp
+		} else {
+			// Use btree as default index type.
```
Comment (zimulala) [inline ddl/index.go]: Update the comment.

**Comment by zimulala** [comment]
Rest LGTM

File: executor/show_test.go [hunk 2]
```diff
@@ -78,10 +78,22 @@ func (s *testSuite) TestShow(c *C) {
...
+	tk.MustExec(`create index idx6 using hash on show_index (id);`)
 	testSQL = "SHOW index from show_index;"
 	tk.MustQuery(testSQL).Check(testutil.RowsWithSep("|",
 		"show_index|0|PRIMARY|1|id|utf8_bin|0|<nil>|<nil>||BTREE||",
 		"show_index|1|cIdx|1|c|utf8_bin|0|<nil>|<nil>|YES|HASH||index_comment_for_cIdx",
+		"show_index|1|idx1|1|id|utf8_bin|0|<nil>|<nil>|YES|HASH||",
+		"show_index|1|idx2|1|id|utf8_bin|0|<nil>|<nil>|YES|||idx",
+		"show_index|1|idx3|1|id|utf8_bin|0|<nil>|<nil>|YES|HASH||idx",
+		"show_index|1|idx4|1|id|utf8_bin|0|<nil>|<nil>|YES|BTREE||idx",
+		"show_index|1|idx5|1|id|utf8_bin|0|<nil>|<nil>|YES|HASH||idx",
```
Comment (XuHuaiyu) [inline executor/show_test.go]: here should be `btree`?

Comment (jackysp) [inline executor/show_test.go]: yes, fixed.

File: ddl/index.go (hunk 1 above)
Comment (jackysp) [inline ddl/index.go]: Done.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**